tests/cxx/default_tpl_arg.cc

```diff
@@ -43,6 +43,25 @@ struct Outer2 {
 // IWYU: IndirectTemplate needs a declaration
 Outer2<int, IndirectTemplate<int>> o2;
 
+void Fn() {
+  // IWYU: FnWithDefaultTplArg is...*default_tpl_arg-i1.h
+  // IWYU: IndirectClass is...*indirect.h
```

Comment:
So are these attributed here because of the reexport hint (`class IndirectClass;`) before the definition? I think it would be nice to show the "default" behavior as well, where `IndirectClass` should be attributed to the template file... right?

I was playing with graveljp's example here: https://github.com/graveljp/iwyu-bugs/tree/main/default_template_param, but it still reports the default template type at point of instantiation.

Reply:
> I was playing with graveljp's example here: https://github.com/graveljp/iwyu-bugs/tree/main/default_template_param, but it still reports the default template type at point of instantiation.

Fixed in #1403. Probably, the test case there is sufficient? Or there should be some without normal default arguments, just with template default arguments?

Reply:
Could you add a comment in the test explaining why `IndirectClass` is required at point of instantiation? It looks wrong to me, unless it's due to the forward-decl near definition.

Once I understand that I'll be in a better position to think about more tests :)

Reply:
In fact, instantiation-side analysis doesn't consider forward declarations currently. That's the reason for redundancy on the first IWYU pass when template definition header neither `#include`s nor forward-declares default template argument type. It is suggested for full type info inclusion on the both sides.